api/internal/plugins/builtinconfig/transformerconfig.go

```diff
@@ -32,14 +33,38 @@ func MakeEmptyConfig() *TransformerConfig {
 	return &TransformerConfig{}
 }
 
+func (t *TransformerConfig) DeepCopy() *TransformerConfig {
+	return &TransformerConfig{
+		NamePrefix:        t.NamePrefix.DeepCopy(),
+		NameSuffix:        t.NameSuffix.DeepCopy(),
+		NameSpace:         t.NameSpace.DeepCopy(),
+		CommonLabels:      t.CommonLabels.DeepCopy(),
+		TemplateLabels:    t.TemplateLabels.DeepCopy(),
+		CommonAnnotations: t.CommonAnnotations.DeepCopy(),
+		NameReference:     t.NameReference.DeepCopy(),
+		VarReference:      t.VarReference.DeepCopy(),
+		Images:            t.Images.DeepCopy(),
+		Replicas:          t.Replicas.DeepCopy(),
+	}
+}
+
+var (
+	initDefaultConfig sync.Once
+	defaultConfig     *TransformerConfig
```

Comment:
Just a note for us: normally a global var raises some alarm bells, but it seems that in this case, this is always populated by the call to `makeTransformerConfigFromBytes(builtinpluginconsts.GetDefaultFieldSpecs())`, which always returns the same value, so this is effectively a const.

Reply:
Thanks, added a note here to the code too. Ideally, it could be avoided. I thought about a few alternatives:
* Generating the data structures as code instead: Would not help much, as DeepCopy would still have to be implemented
* Using gopkg.in/yaml.v3 instead of k8s yaml: Helps a bit not very much
* Using gob/protobuf instead of YAML. Complex?
* loading this data in main instead and passing it as a reference: Still needs DeepCopy. Significant code changes and questionable gains

k8s deepcopy gen: Might be helpful as a future improvement, but I did not want to introduce another compile time dependency without discussion

If you have any other ideas I can try it out